aten/src/ATen/native/native_functions.yaml

```diff
@@ -12765,6 +12765,11 @@
     NestedTensorCPU: NestedTensor_to_padded_tensor_generic
     NestedTensorCUDA: NestedTensor_to_padded_tensor_cuda
 
+- func: _nested_tensor_softmax_with_shape(Tensor self, Tensor query) -> Tensor
```

Comment:
My immediate reaction to this is that, NestedTensor_softmax_dropout is returning a view of the self, and therefore this is an incorrect schema. When this PR lands: https://github.com/pytorch/pytorch/pull/83371, this function would be composite implicit and instead of getting the buffer with `get_buffer` it would call `to_buffer`. I think the indexing loop will correctly register the backward formulas, but thats the only thing I would want to confirm with some grad_check tests.